subsys/bluetooth/host/keys.c

```diff
@@ -509,3 +509,16 @@ void bt_keys_show_sniffer_info(struct bt_keys *keys, void *data)
 	}
 }
 #endif /* defined(CONFIG_BT_LOG_SNIFFER_INFO) */
+
+#ifdef ZTEST_UNITTEST
+struct bt_keys *bt_keys_get_key_pool(void)
```

Comment:
As a matter of interface testing (notwithstanding the previous comments about risks), this should be a pointer to const or const pointer to const to force an explicit cast to enable dangerous manipulation.

Reply:
The point is that some tests need to explicitly clear the list as if it is a fresh start, modifying the pointer to point to constant data will prevent such operation.